devito/symbolics/manipulation.py

```diff
@@ -157,16 +156,21 @@ def xreplace_indices(exprs, mapper, key=None, only_rhs=False):
 
 def pow_to_mul(expr):
     if expr.is_Atom or expr.is_Indexed:
-        return expr
+        out = expr
     elif expr.is_Pow:
         base, exp = expr.as_base_exp()
         if exp <= 0 or not exp.is_integer:
             # Cannot handle powers containing non-integer non-positive exponents
-            return expr
+            out = expr
         else:
-            return sympy.Mul(*[base]*exp, evaluate=False)
+            out = Mul(*[base]*exp, evaluate=False)
     else:
-        return expr.func(*[pow_to_mul(i) for i in expr.args], evaluate=False)
+        out = expr.func(*[pow_to_mul(i) for i in expr.args], evaluate=False)
+
+    if isinstance(expr, Differentiable):
```

Comment:
hack. At this point, ``expr`` should already be either a Differentiable or a pure expr.